core/structure.py

```diff
@@ -436,7 +436,7 @@ class ReadingsWeatherClass(BASE):
     )
 
     timestamp = Column(DateTime, nullable=False, index=True)
-    temperature = Column(Float, nullable=False)
+    temperature = Column(Float, nullable=True)
```

Comment:
This one might have actually been just for consistency, since all the other comparable columns were nullable. Or maybe there also was something in the old database, not sure.